### PR TITLE
{lib}[foss/2018a] Trilinos v12.12.1 w/ Python 3.6.4 (+ deps)

### PR DESCRIPTION
**Pull request by boegel**
(created using `eb --new-pr`)

edit: requires ~~https://github.com/easybuilders/easybuild-easyblocks/pull/1486~~

### CODE REVIEW

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 4 out of 5 (5 easyconfigs in this PR)
node3131.skitty.os - Linux centos linux 7.5.1804, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 2.7.5
See https://gist.github.com/341547331c54b565574b64648cd93b00 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 5 out of 5 (5 easyconfigs in this PR)
node3131.skitty.os - Linux centos linux 7.5.1804, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 2.7.5
See https://gist.github.com/01753ff43e5233d4da48b6a988842a8a for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 5 out of 5 (5 easyconfigs in this PR)
node2716.swalot.os - Linux centos linux 7.5.1804, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/a65e8e82983cb87210108c9dd923e699 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 5 out of 5 (5 easyconfigs in this PR)
node2046.delcatty.os - Linux centos linux 7.5.1804, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/961276d9b8f3325b933864950eb3d372 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 5 out of 5 (5 easyconfigs in this PR)
node2033.delcatty.os - Linux centos linux 7.5.1804, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/06f3e2b0a2b4dadce62daa7fa76a1e32 for a full test report.

File: easybuild/easyconfigs/t/Trilinos/Trilinos-12.12.1-foss-2018a-Python-3.6.4.eb [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+homepage = 'https://trilinos.org'
+description = """The Trilinos Project is an effort to develop algorithms and enabling technologies
+ within an object-oriented software framework for the solution of large-scale, complex multi-physics
+ engineering and scientific problems. A unique design feature of Trilinos is its focus on packages."""
+
+toolchain = {'name': 'foss', 'version': '2018a'}
+toolchainopts = {'usempi': True, 'pic': True, 'strict': True}
+
+source_urls = ['http://trilinos.csbsju.edu/download/files/']
```
Comment (vanzod) [inline easybuild/easyconfigs/t/Trilinos/Trilinos-12.12.1-foss-2018a-Python-3.6.4.eb]: This URL is (maybe temporarily) unavailable. How about using the release archives in their GitHub repo?

Comment (boegel) [reply easybuild/easyconfigs/t/Trilinos/Trilinos-12.12.1-foss-2018a-Python-3.6.4.eb]: I don't think the tarball available via https://github.com/trilinos/Trilinos/releases is complete. I ran a `diff -ru`, and got back a whole bunch of `Only in ...: xyz` messages.

Let's see if we get any replies here: https://github.com/trilinos/Trilinos/issues/3447

Comment (boegel) [reply easybuild/easyconfigs/t/Trilinos/Trilinos-12.12.1-foss-2018a-Python-3.6.4.eb]: Found an alternate (probably the correct) location via our dear friend Google: https://trilinos.org/oldsite/download/files/, so I'll add that as a fallback URL.

**Comment by vanzod** [comment]
Test report by @vanzod
**FAILED**
Build succeeded for 3 out of 4 (4 easyconfigs in this PR)
cermis - Linux debian 9.5, Intel(R) Xeon(R) CPU           E5620  @ 2.40GHz, Python 2.7.13
See https://gist.github.com/2c3ef4723f24b4a67b704d0307142394 for a full test report.

**Comment by vanzod** [comment]
It looks like there is a certificate issue when attempting to download from the `oldsite` URL:
```
filetools.py:500 WARNING IOError occurred while trying to download https://trilinos.org/oldsite/download/files/trilinos-12.12.1-Source.tar.gz to /opt/easybuild/sources/t/Trilinos/trilinos-12.12.1-Source.tar.gz: <urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:661)>
```
The only workaround seem to be skipping the certificate verification by modifying the `urllib.urlopen` call in https://github.com/vanzod/easybuild-framework/blob/c4d2ea3b2874d68f6655aebc0580cf194bb06702/easybuild/tools/filetools.py#L354 as:
```
context = ssl._create_unverified_context()
url_fd = urllib2.urlopen(url_req, timeout=timeout, context=context)
```

**Comment by vanzod** [comment]
Test report by @vanzod
**FAILED**
Build succeeded for 3 out of 4 (4 easyconfigs in this PR)
cermis - Linux debian 9.5, Intel(R) Xeon(R) CPU           E5620  @ 2.40GHz, Python 2.7.13
See https://gist.github.com/6f2f04112cc6015c919f77a991590775 for a full test report.

**Comment by vanzod** [comment]
`Trilinos` needs `X11` for `TPL` unless we pass `-DTPL_ENABLE_X11=OFF`

**Comment by vanzod** [comment]
Test report by @vanzod
**FAILED**
Build succeeded for 3 out of 4 (4 easyconfigs in this PR)
cermis - Linux debian 9.5, Intel(R) Xeon(R) CPU           E5620  @ 2.40GHz, Python 2.7.13
See https://gist.github.com/e4419ff4ff400e8b0417f63f61cf2c7b for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
node3138.skitty.os - Linux centos linux 7.5.1804, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 2.7.5
See https://gist.github.com/c52093a869be949174a5aaa2db5718cc for a full test report.

**Comment by boegel** [comment]
@vanzod It looks like you're not using the updated easyblock from https://github.com/easybuilders/easybuild-easyblocks/pull/1486?

```
/opt/easybuild/software/MPI/GCC/6.4.0-2.28/OpenMPI/2.1.2/SuiteSparse/5.1.2-METIS-5.1.0/CHOLMOD/Lib/libcholmod.a(cholmod_metis.o):cholmod_metis.c:function cholmod_metis_bisector: error: undefined reference to 'METIS_ComputeVertexSeparator'
/opt/easybuild/software/MPI/GCC/6.4.0-2.28/OpenMPI/2.1.2/SuiteSparse/5.1.2-METIS-5.1.0/CHOLMOD/Lib/libcholmod.a(cholmod_metis.o):cholmod_metis.c:function cholmod_metis: error: undefined reference to 'METIS_NodeND'
```

**Comment by vanzod** [comment]
Test report by @vanzod
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
cermis - Linux debian 9.5, Intel(R) Xeon(R) CPU           E5620  @ 2.40GHz, Python 2.7.13
See https://gist.github.com/32063e9c1c252654efa10801351a27d9 for a full test report.

**Comment by vanzod** [comment]
Going in, thanks @boegel!

**Comment by boegel** [comment]
\o/

**Pull request merged**